### PR TITLE
[LLM] Add more transformers int4 examples (Falcon)

### PR DESCRIPTION
**Pull request by Oscilloscope98**
## Description

Add model Falcon to transformers int4 examples

### 1. Why the change?

Add more model-specific examples so that users can quickly get started using BigDL-LLM to run some popular open-source models in the community

### 2. User API changes

N/A

### 3. Summary of the change 

Add transformers int4 generate examples for Falcon( scripts & README & patched modelling_RW.py)

### 4. How to test?
- [x] Local test on Client
- [x] Local test on Server (falcon-40b only on server)

### CODE REVIEW

File: python/llm/example/transformers/transformers_int4/falcon/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+pip install bigdl-llm[all] # install bigdl-llm with 'all' option
+pip install einops # additional package required for falcon-7b-instruct and falcon-40b-instruct to conduct generation
+```
+
+### 2. (Optional) Download Model and Replace File
+If you select the Falcon models [tiiuae/falcon-7b-instruct](https://huggingface.co/tiiuae/falcon-7b-instruct) or [tiiuae/falcon-40b-instruct](https://huggingface.co/tiiuae/falcon-40b-instruct), please note that there is a bug in their model implementation (`modelling_RW.py`) that causes failures when using the KV cache. Consequently, the BigDL-LLM INT4 optimization may not achieve optimal performance with these models.
+
+To address this issue, we have provided two files: [src/falcon-7b-instruct/modelling_RW.py](./src/falcon-7b-instruct/modelling_RW.py) and [src/falcon-40b-instruct/modelling_RW.py](./src/falcon-40b-instruct/modelling_RW.py). You can replace the `modelling_RW.py file` in the `tiiuae/falcon-7b-instruct` or `tiiuae/falcon-40b-instruct` model files respectively with these files. By doing so, you can resolve the KV cache bug and obtain optimal performance from the BigDL-LLM INT4 optimization.
+
```
Comment (jason-dai) [inline python/llm/example/transformers/transformers_int4/falcon/README.md:24]: If you select the Falcon models ([tiiuae/falcon-7b-instruct](https://huggingface.co/tiiuae/falcon-7b-instruct) or [tiiuae/falcon-40b-instruct](https://huggingface.co/tiiuae/falcon-40b-instruct)), please note that their code (`modelling_RW.py`) does not support KV cache at the moment. To address issue, we have provided two updated files ([falcon-7b-instruct/modelling_RW.py](./src/falcon-7b-instruct/modelling_RW.py) and [falcon-40b-instruct/modelling_RW.py](./src/falcon-40b-instruct/modelling_RW.py)), which can be used to achieve the best performance using BigDL-LLM INT4 optimizations with KV cache support.

File: python/llm/example/transformers/transformers_int4/falcon/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+#### 2.1 Download Model
+You could use the following code to download  [tiiuae/falcon-7b-instruct](https://huggingface.co/tiiuae/falcon-7b-instruct) or [tiiuae/falcon-40b-instruct](https://huggingface.co/tiiuae/falcon-40b-instruct) with a specific snapshot id. Please note that the provided patched `modelling_RW.py` files are specifically based on these commits.
```
Comment (jason-dai) [inline python/llm/example/transformers/transformers_int4/falcon/README.md:26]: Please note that the `modelling_RW.py` files that we provide are  based on these specific commits

File: python/llm/example/transformers/transformers_int4/falcon/README.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                               revision="1e7fdcc9f45d13704f3826e99937917e007cd975",
+                               cache_dir="dir/path/where/model/is/stored")
+print(f'tiiuae/falcon-40b-instruct is downloaded to {model_path}')
+```
+
+#### 2.2 Replace `modelling_RW.py`
+For `tiiuae/falcon-7b-instruct`, you should replace the `modelling_RW.py` with [src/falcon-7b-instruct/modelling_RW.py](./src/falcon-7b-instruct/modelling_RW.py).
+
+For `tiiuae/falcon-40b-instruct`, you should replace the `modelling_RW.py` with [src/falcon-40b-instruct/modelling_RW.py](./src/falcon-40b-instruct/modelling_RW.py).
+
```
Comment (jason-dai) [inline python/llm/example/transformers/transformers_int4/falcon/README.md:48]: For `tiiuae/falcon-7b-instruct`, you should replace the `modelling_RW.py` with [falcon-7b-instruct/modelling_RW.py](./src/falcon-7b-instruct/modelling_RW.py).

For `tiiuae/falcon-40b-instruct`, you should replace the `modelling_RW.py` with [falcon-40b-instruct/modelling_RW.py](./src/falcon-40b-instruct/modelling_RW.py).

**Review by cyita — APPROVED** [review]
LGTM

**Review by plusbang — APPROVED** [review]
LGTM

File: python/llm/example/transformers/transformers_int4/falcon/README.md (hunk 1 above)
Comment (Oscilloscope98) [reply python/llm/example/transformers/transformers_int4/falcon/README.md:23]: Thanks for the comments. I have updated the README :)

**Comment by Oscilloscope98** [comment]
@jason-dai Merged for now. Will raise another PR for any new comments :)

**Pull request merged**